{
	"document": {
		"aggregate_severity": {
			"namespace": "https://access.redhat.com/security/updates/classification/",
			"text": "Moderate"
		},
		"category": "csaf_security_advisory",
		"csaf_version": "2.0",
		"distribution": {
			"text": "Copyright Â© Red Hat, Inc. All rights reserved.",
			"tlp": {
				"label": "WHITE",
				"url": "https://www.first.org/tlp/"
			}
		},
		"lang": "en",
		"notes": [
			{
				"category": "summary",
				"text": "Red Hat Ansible Tower 3.6 runner release (CVE-2019-18874)",
				"title": "Topic"
			},
			{
				"category": "general",
				"text": "* Updated python-psutil version to 5.6.6 inside ansible-runner container\n(CVE-2019-18874)",
				"title": "Details"
			},
			{
				"category": "legal_disclaimer",
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use"
			}
		],
		"publisher": {
			"category": "vendor",
			"contact_details": "https://access.redhat.com/security/team/contact/",
			"issuing_authority": "Red Hat Product Security is responsible for vulnerability handling across all Red Hat products and services.",
			"name": "Red Hat Product Security",
			"namespace": "https://www.redhat.com"
		},
		"references": [
			{
				"category": "self",
				"summary": "https://access.redhat.com/errata/RHSA-2020:4255",
				"url": "https://access.redhat.com/errata/RHSA-2020:4255"
			},
			{
				"category": "external",
				"summary": "https://access.redhat.com/security/updates/classification/#moderate",
				"url": "https://access.redhat.com/security/updates/classification/#moderate"
			},
			{
				"category": "external",
				"summary": "1772014",
				"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1772014"
			},
			{
				"category": "self",
				"summary": "Canonical URL",
				"url": "https://security.access.redhat.com/data/csaf/v2/advisories/2020/rhsa-2020_4255.json"
			}
		],
		"title": "Red Hat Security Advisory: security update - Red Hat Ansible Tower 3.6 runner release (CVE-2019-18874)",
		"tracking": {
			"current_release_date": "2024-11-13T22:15:51+00:00",
			"generator": {
				"date": "2024-11-13T22:15:51+00:00",
				"engine": {
					"name": "Red Hat SDEngine",
					"version": "4.2.0"
				}
			},
			"id": "RHSA-2020:4255",
			"initial_release_date": "2020-10-14T13:09:19+00:00",
			"revision_history": [
				{
					"date": "2020-10-14T13:09:19+00:00",
					"number": "1",
					"summary": "Initial version"
				},
				{
					"date": "2020-10-14T13:09:19+00:00",
					"number": "2",
					"summary": "Last updated version"
				},
				{
					"date": "2024-11-13T22:15:51+00:00",
					"number": "3",
					"summary": "Last generated version"
				}
			],
			"status": "final",
			"version": "3"
		}
	},
	"product_tree": {
		"branches": [
			{
				"branches": [
					{
						"branches": [
							{
								"category": "product_name",
								"name": "Red Hat Ansible Tower 3.6 for RHEL 7",
								"product": {
									"name": "Red Hat Ansible Tower 3.6 for RHEL 7",
									"product_id": "7Server-Ansible-Tower-3.6",
									"product_identification_helper": {
										"cpe": "cpe:/a:redhat:ansible_tower:3.6::el7"
									}
								}
							}
						],
						"category": "product_family",
						"name": "Red Hat Ansible Tower"
					},
					{
						"branches": [
							{
								"category": "product_version",
								"name": "ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64",
								"product": {
									"name": "ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64",
									"product_id": "ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64",
									"product_identification_helper": {
										"purl": "pkg:oci/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158?arch=amd64&repository_url=registry.redhat.io/ansible-tower-36/ansible-runner-rhel7&tag=1.4.4-2"
									}
								}
							}
						],
						"category": "architecture",
						"name": "amd64"
					}
				],
				"category": "vendor",
				"name": "Red Hat"
			}
		],
		"relationships": [
			{
				"category": "default_component_of",
				"full_product_name": {
					"name": "ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64 as a component of Red Hat Ansible Tower 3.6 for RHEL 7",
					"product_id": "7Server-Ansible-Tower-3.6:ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64"
				},
				"product_reference": "ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64",
				"relates_to_product_reference": "7Server-Ansible-Tower-3.6"
			}
		]
	},
	"vulnerabilities": [
		{
			"acknowledgments": [
				{
					"names": [
						"Riccardo Schirone"
					],
					"organization": "Red Hat",
					"summary": "This issue was discovered by Red Hat."
				}
			],
			"cve": "CVE-2019-18874",
			"cwe": {
				"id": "CWE-416",
				"name": "Use After Free"
			},
			"discovery_date": "2019-11-13T00:00:00+00:00",
			"ids": [
				{
					"system_name": "Red Hat Bugzilla ID",
					"text": "1772014"
				}
			],
			"notes": [
				{
					"category": "description",
					"text": "A double free issue has been discovered in python-psutil because of the mishandling of refcounts while converting system data into Python objects in functions like psutil_disk_partitions(), psutil_users(), psutil_net_if_addrs(), and others. In particular cases, a local attacker may be able to get code execution by manipulating system resources that python-psutil then tries to convert.",
					"title": "Vulnerability description"
				},
				{
					"category": "summary",
					"text": "python-psutil: Double free because of refcount mishandling",
					"title": "Vulnerability summary"
				},
				{
					"category": "general",
					"text": "The CVSS score(s) listed for this vulnerability do not reflect the associated product's status, and are included for informational purposes to better understand the severity of this vulnerability.",
					"title": "CVSS score applicability"
				}
			],
			"product_status": {
				"fixed": [
					"7Server-Ansible-Tower-3.6:ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64"
				]
			},
			"references": [
				{
					"category": "self",
					"summary": "Canonical URL",
					"url": "https://access.redhat.com/security/cve/CVE-2019-18874"
				},
				{
					"category": "external",
					"summary": "RHBZ#1772014",
					"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1772014"
				},
				{
					"category": "external",
					"summary": "https://www.cve.org/CVERecord?id=CVE-2019-18874",
					"url": "https://www.cve.org/CVERecord?id=CVE-2019-18874"
				},
				{
					"category": "external",
					"summary": "https://nvd.nist.gov/vuln/detail/CVE-2019-18874",
					"url": "https://nvd.nist.gov/vuln/detail/CVE-2019-18874"
				}
			],
			"release_date": "2019-11-07T00:00:00+00:00",
			"remediations": [
				{
					"category": "vendor_fix",
					"date": "2020-10-14T13:09:19+00:00",
					"details": "For information on upgrading Ansible Tower, reference the Ansible Tower Upgrade and Migration Guide: https://docs.ansible.com/ansible-tower/latest/html/upgrade-migration-guide/index.html",
					"product_ids": [
						"7Server-Ansible-Tower-3.6:ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64"
					],
					"restart_required": {
						"category": "none"
					},
					"url": "https://access.redhat.com/errata/RHSA-2020:4255"
				}
			],
			"scores": [
				{
					"cvss_v3": {
						"attackComplexity": "HIGH",
						"attackVector": "LOCAL",
						"availabilityImpact": "HIGH",
						"baseScore": 7,
						"baseSeverity": "HIGH",
						"confidentialityImpact": "HIGH",
						"integrityImpact": "HIGH",
						"privilegesRequired": "NONE",
						"scope": "UNCHANGED",
						"userInteraction": "REQUIRED",
						"vectorString": "CVSS:3.0/AV:L/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H",
						"version": "3.0"
					},
					"products": [
						"7Server-Ansible-Tower-3.6:ansible-tower-36/ansible-runner-rhel7@sha256:332821e5b50d1570d6a3be7c3ea76d7f1d84d6754b467db49ed5d2d753a1f158_amd64"
					]
				}
			],
			"threats": [
				{
					"category": "impact",
					"details": "Moderate"
				}
			],
			"title": "python-psutil: Double free because of refcount mishandling"
		}
	]
}
